### PR TITLE
Do not pollute home dir

### PR DESCRIPTION
**Pull request by mmlb**
Lets be good cli folks and not pollute a user's $HOME, instead lets put config files in platform specified config paths. ~/.mctl.yml is used if it exists otherwise we look in dirs provided by xdg package.

### CODE REVIEW

**Comment by mmlb** [comment]
This is ready to go but I made it draft until #20 is merged as this depends on it.

**Review by joelrebel — APPROVED** [review]
LGTM

File: cmd/root.go [hunk 1]
```diff
@@ -45,6 +45,6 @@ func Execute() {
 }
 
 func init() {
-	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.mctl.yml)")
+	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $XDG_CACHE_DIR/mctl/config.yml)")
```
Comment (diogomatsubara) [inline cmd/root.go:48]: Shouldn't this be `$XDG_CONFIG_HOME` instead? According to the spec https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html, it's where the user config files should be located. This env var wouldn't be used on a Mac environment, AFAICT.

Comment (mmlb) [reply cmd/root.go:48]: 1000% right... will fix

Comment (mmlb) [reply cmd/root.go]: > This env var wouldn't be used on a Mac environment, AFAICT.

The go package I'm using here basically does the right thing for other platforms too:

#### Unix-like operating systems
||Unix|macOS|Plan 9|
|--|--|--|--|
|XDG_CONFIG_HOME|~/.config|~/Library/Application Support|$home/lib|

#### Microsoft Windows
||Known Folder(s)|Fallback(s)|
|--|--|--|
|XDG_CONFIG_HOME|LocalAppData|%LOCALAPPDATA%|

Comment (diogomatsubara) [reply cmd/root.go]: Not quite right still, diff shows `$XDG_CONFIG_DIR` while it should be `$XDG_CONFIG_HOME`. 

Good to know it works everywhere! Thanks for the fixes!

Comment (mmlb) [reply cmd/root.go]: thats what I get when I sed w/o thinking :/

Comment (mmlb) [reply cmd/root.go]: @diogomatsubara ptal again, thanks!

**Review by diogomatsubara — APPROVED** [review]

**Comment by mmlb** [comment]
:tada:

**Comment by mmlb** [comment]
uggh I *very much dislike* that merge commits aren't allowed in this org

**Pull request merged**